runtime/src/main/java/org/corfudb/protocols/wireprotocol/LogData.java

```diff
@@ -174,9 +173,6 @@ public LogData(DataType type, final Object object) {
             this.type = type;
             this.data = null;
             this.payload.set(object);
-            if (object instanceof LogEntry) {
-                ((LogEntry) object).setEntry(this);
```

Comment:
With this patch, we cannot setGlobalAddress here because by the time LogData is constructed, the address (token) is not yet assigned.